qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/jdt/internal/template/TemplateGenerateMissingJavaMember.java

```diff
@@ -378,9 +384,25 @@ private static WorkspaceEdit createNewTemplateExtensionFile(GenerateMissingJavaM
 
 		IPackageFragment destPackage = null;
 		try {
+			IPath sourceFilesPath = project.getResource().getProjectRelativePath().append("src/main/java");
```

Comment:
²I wonder ifyou use IJavaProject#getRawClasspath should be easier? The Eclipse IDE wizard of new class initialize the source folder 

![image](https://user-images.githubusercontent.com/1932211/228396030-7ca8af93-b55c-4bc8-8a1d-d74dd9318ab3.png)


with this code

https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/7caebb6cbb2167d1b5e33f4f8ba31273c47f75a5/org.eclipse.jdt.ui/ui/org/eclipse/jdt/ui/wizards/NewContainerWizardPage.java#L251

Reply:
Unfortunately, this has the issue that it sometimes returns `src/main/resources` instead of `src/main/java`

Reply:
After playing with the .classpath file, the Eclipse IDE wizard shows src/main/resources when src/main/resources is declared in the classpath before the src/main/java but is it a commons usecase?

Reply:
well you can iterate over javaProject.getRawClasspath(), filter on CPE_SOURCE, and return the source where  path == src/main/java